### PR TITLE
Update org.slf4j dependency

### PR DESCRIPTION
**Pull request by sophiejiangg**
# Description

| **Story ID** | **Story Name** |
| ------------ | -------------- |
| #174663023 |  Update org.slf4j dependency |

***
## Summary of Changes: 
Bring in slf4j-api as a dependency instead of slf4j-log4j12

***
## Review readiness
This pull request is considered a _Work In Progress_ unless all of the following boxes are checked.
Items not required for this PR should be removed from the below list.

- [x] A secret scan was run:
    - [x] on every commit
    - [x] on every push
- [x] Existing tests of the following types were run:
    - [x] unit tests
    - [x] integration tests
- [x] Application was run to ensure it is still working as expected
- [x] Examples given to customers still work as expected
- [x] Code quality gates have been meet:
    - [x] Code Coverage
    - [x] Sonar Quality Gate     
- [x] Ensured no breaking changes to the public interface
- [ ] One of the following has been done:
    - [ ] Documentation was updated
    - [ ] A story was created to update the documentation

***
## PR Reviewer's Checklist
Be sure to give yourself enough time to review the PR.  We are looking for good approvals, not rubber stamped approvals.
- [ ] Pull down the branch
- [ ] Review the code
- [ ] Build the code
- [ ] Run tests, application and/or examples locally
- [ ] Review code's coverage
- [ ] Review code's Sonar quality gate

***
## Reviewers for this PR: 
- [ ] @autonomic-tmc/sdk-team
- [ ] @autonomic-tmc/fe-pr-reviewers

### CODE REVIEW

File: src/test/java/com/autonomic/tmc/auth/ClientCredentialsTokenSupplierTest.java [hunk 1]
```diff
@@ -41,7 +41,6 @@
 import java.util.concurrent.TimeUnit;
 import java.util.stream.Stream;
 import lombok.extern.slf4j.Slf4j;
-import org.apache.log4j.BasicConfigurator;
```
Comment (shah-jenish) [inline src/test/java/com/autonomic/tmc/auth/ClientCredentialsTokenSupplierTest.java:44]: **This comment is NOT related to this line or the changes in scope of this PR**

I noticed the following WARNING in Maven output:
```
[WARNING] /Users/jenishshah/code/autonomic-tmc/tmc-auth/src/test/java/com/autonomic/tmc/auth/ClientCredentialsTokenSupplierTest.java: /Users/jenishshah/code/autonomic-tmc/tmc-auth/src/test/java/com/autonomic/tmc/auth/ClientCredentialsTokenSupplierTest.java uses unchecked or unsafe operations.
```

The fix for this Warning is to change method signature as follows:

```java
    <T extends Throwable> void anyExceptionThrown_alwaysRethrowsAsSdkException(String expectedMessage, Throwable cause,
        Class<T> expectedExceptionOfType) {
```

File: THIRD-PARTY.txt [hunk 2]
```diff
@@ -34,8 +34,6 @@ Date Generated: 2021-03-11 15:17
  (Apache License, Version 2.0) Apache Commons Logging (commons-logging:commons-logging:1.2 - http://commons.apache.org/proper/commons-logging/)
  (CDDL 1.1) JavaBeans Activation Framework (JAF) (javax.activation:activation:1.1 - http://java.sun.com/products/javabeans/jaf/index.jsp)
  (CDDL 1.1) Java Servlet API (javax.servlet:javax.servlet-api:3.1.0 - http://servlet-spec.java.net)
- (CDDL 1.1) (GPL2 w/ CPE) jaxb-api (javax.xml.bind:jaxb-api:2.3.0 - https://github.com/javaee/jaxb-spec/jaxb-api)
- (Apache License, Version 2.0) Apache Log4j (log4j:log4j:1.2.17 - http://logging.apache.org/log4j/1.2/)
```
Comment (shah-jenish) [inline THIRD-PARTY.txt:38]: 👍🏽

File: THIRD-PARTY.txt [hunk 3]
```diff
@@ -76,7 +74,6 @@ Date Generated: 2021-03-11 15:17
  (BSD 3-Clause) asm (org.ow2.asm:asm:7.0 - http://asm.ow2.org/)
  (MIT-License) Project Lombok (org.projectlombok:lombok:1.18.0 - https://projectlombok.org)
  (MIT-License) SLF4J API Module (org.slf4j:slf4j-api:1.7.30 - http://www.slf4j.org)
- (MIT-License) SLF4J LOG4J-12 Binding (org.slf4j:slf4j-log4j12:1.7.30 - http://www.slf4j.org)
```
Comment (shah-jenish) [inline THIRD-PARTY.txt:79]: 👍🏽

**Review by shah-jenish — CHANGES_REQUESTED** [review]
The requested change is unrelated to the work done in this PR. It addresses one Warning in Maven output.

Changes look good. I saw the failure in `tmc-feed-consumer` using the released version of `tmc-auth` without excluding the `log4j` dependency. When using the local version of `tmc-auth`, failure went away. I did not have to exclude the `log4j` dependency as expected.

**Review by shah-jenish — APPROVED** [review]
🚀

**Review by cbusby — APPROVED** [review]
